tests/lib/prepare.sh

```diff
@@ -518,3 +529,266 @@ prepare_all_snap() {
 }
+
+setup_reflash_magic_core18() {
+        # install the stuff we need
+        distro_install_package kpartx busybox-static
+        distro_install_local_package "$GOHOME"/snapd_*.deb
+        distro_clean_package_cache
+
+        snap install "--${CORE_CHANNEL}" core18
+        snap install "--${CORE_CHANNEL}" snapd
+        # install ubuntu-image
+        snap install --classic --edge ubuntu-image
+
+        # needs to be under /home because ubuntu-device-flash
+        # uses snap-confine and that will hide parts of the hostfs
+        IMAGE_HOME=/home/image
+        mkdir -p "$IMAGE_HOME"
+
+        # modify the snapd snap so that it has our snapd
+        UNPACKD="/tmp/snapd-snap"
+        unsquashfs -d "$UNPACKD" /var/lib/snapd/snaps/snapd_*.snap
+        dpkg-deb -x "$SPREAD_PATH"/../snapd_*.deb "$UNPACKD"
+        snap pack "$UNPACKD" "$IMAGE_HOME"
+
+        # FIXME: fetch directly once its in the assertion service
+        cp "$TESTSLIB/assertions/core-amd64-18.model" "$IMAGE_HOME/pc.model"
+
+        IMAGE=core18-amd64.img
+
+        # ensure that ubuntu-image is using our test-build of snapd with the
+        # test keys and not the bundled version of usr/bin/snap from the snap.
+        # Note that we can not put it into /usr/bin as '/usr' is different
+        # when the snap uses confinement.
+        cp /usr/bin/snap "$IMAGE_HOME"
+        export UBUNTU_IMAGE_SNAP_CMD="$IMAGE_HOME/snap"
+
+        EXTRA_FUNDAMENTAL=
+        IMAGE_CHANNEL=edge
+        if [ "$KERNEL_CHANNEL" = "$GADGET_CHANNEL" ]; then
+            IMAGE_CHANNEL="$KERNEL_CHANNEL"
+        else
+            # download pc-kernel snap for the specified channel and set ubuntu-image channel
+            # to gadget, so that we don't need to download it
+            snap download --channel="$KERNEL_CHANNEL" pc-kernel
+
+            EXTRA_FUNDAMENTAL="--extra-snaps $PWD/pc-kernel_*.snap"
+            IMAGE_CHANNEL="$GADGET_CHANNEL"
+        fi
+
+        # TODO: once we have a real "canonical" signed core18 model
+        # use this and remove the extra assertions setup below.
+        #
+        # Note we don't need teardown, once the image is written
+        # the system reboots anyway.
+        #
+        # We can do this one https://forum.snapcraft.io/t/5947 is
+        # answered.
+        echo "Added needed assertions so that core-amd64-18.model works"
+        . "$TESTSLIB/store.sh"
+        export STORE_DIR="$(pwd)/fake-store-blobdir"
+        export STORE_ADDR="localhost:11028"
+        setup_fake_store "$STORE_DIR"
+        cp "$TESTSLIB"/assertions/developer1.account "$STORE_DIR"/asserts
+        cp "$TESTSLIB"/assertions/developer1.account-key "$STORE_DIR"/asserts
+        # have snap use the fakestore for assertions (but nothing else)
+        export SNAPPY_FORCE_SAS_URL="http://$STORE_ADDR"
+        # -----------------------8<----------------------------
+
+        /snap/bin/ubuntu-image -w "$IMAGE_HOME" "$IMAGE_HOME/pc.model" \
```

Comment:
We already had some issues with the ubuntu-image snap, that's why we are using the deb version for beta validation. If I remember well there was a problem with squahfs, do you? do you know if it is already fixed?

Reply:
I don't remember the details what the issue was with the snap, sorry. I can look into this in a followup but right now it is not giving us problems AFAICT.